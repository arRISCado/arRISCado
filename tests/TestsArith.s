.global main
.text

main:
	addi a0, zero, 1
	sub a1, zero, a0
	add a2, a0, a0
	add a2, a2, a2
	addi a3, zero, 2047
	add a4, a2, a1
	sub a5, a2, a4
	addi a6, a3, 5
	lui a7, 98
	auipc s2, 0

# Expected final state:
# a0: 1
# a1: 4294967295 (-1)
# a2: 4
# a3: 2047
# a4: 3
# a5: 1
# a6: 2052
# a7: 401408
# s2: 80
