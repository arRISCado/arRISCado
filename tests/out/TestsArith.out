0           x0	zero
x           x1	ra
x           x2	sp
x           x3	gp
x           x4	tp
x           x5	t0
x           x6	t1
x           x7	t2
x           x8	s0
x           x9	s1
1           x10	a0
4294967295  x11	a1
4           x12	a2
2047        x13	a3
3           x14	a4
1           x15	a5
2052        x16	a6
401408      x17	a7
80          x18	s2
x           x19	s3
x           x20	s4
x           x21	s5
x           x22	s6
x           x23	s7
x           x24	s8
x           x25	s9
x           x26	s10
x           x27	s11
x           x28	t3
x           x29	t4
x           x30	t5
x           x31	t6